### ISSUE TITLE
how to change num of classes?

### ISSUE THREAD

**Issue by shubhaminnani** [NVIDIA/semantic-segmentation#24]
https://github.com/NVIDIA/semantic-segmentation/blob/c62756222a6cf61f7950b4f47f26abf275c57615/train.py#L289

Hi, Where to specify this argument highlighted above since parser doesn't have anything about classes. Thank you in advance. I want to change num of classes for other dataset but after training IOU for only some classes appears. Some class IOU is nan. @bryancatanzaro @swarren @aaronp24 @mkolod @lukeyeager

**Comment by shubhaminnani** [comment]
I tried changing it but only some class IoU are getting a number. Rest are NaN

**Comment by bryanyzhu** [comment]
This may depend on your dataset. There are several possible reasons,

- There is a chance that you didn't use the correct label space. For example, like Cityscapes, there are two kinds of annotations, `train_id` and `label`. They are different. 

- There is a chance that your trained model performs bad (due to overfitting). And the model only predicts some classes, while the rest will be NaN.

- There is a chance that your validation set doesn't contain the rest classes, so the IoU is NaN. 

There are multiple reasons, but most likely it is because of the new dataset. You may need to change more than just this [one line](https://github.com/NVIDIA/semantic-segmentation/blob/c62756222a6cf61f7950b4f47f26abf275c57615/datasets/cityscapes.py#L19) to properly train/test your model.

**Comment by karansapra** [comment]
@shubhaminnani can you tell us more about your dataset. That might help debug the problem. Or if you solved the problem, let us know and we can close this issue.

**Comment by shubhaminnani** [comment]
dataset is same as cityscapes having change in num of classes=26. and change in name of gtFine as in cityscapes its labeltrainids, In my dataset its level3ids. Rest is same. Still I am getting only some class IoU and overall IoU is around 50. 

IoU in this code is of training or validation while saving checkpoint?

Thanks in advance!

**Comment by bryanyzhu** [comment]
@shubhaminnani IoU in this code is of validation while saving checkpoint.
